app/gui/src/project-view/util/data/array.ts

```diff
@@ -95,7 +95,7 @@ export function partition<T>(array: Iterable<T>, pred: (elem: T) => boolean): [T
 export function findDifferenceIndex<T>(
   lhs: T[],
   rhs: T[],
-  equals = (a: T, b: T) => a === b,
+  equals = (a: T, b: T) => a === b || (Number.isNaN(a) && Number.isNaN(b)),
```

Comment:
I'm a bit unsure of this. If NaN != NaN, this function should also treat them as not equal by default  IMO. Instead, I would use this comparator in proptest, or configure proptests to not include NaNs (maybe ther's a simple way to do it, not being listing all values?)

However, I won't be blocking this PR because of that.

Reply:
`equals = Object.is` is an option too but... yeah idk. depends what this function is being used for i guess.